### ISSUE TITLE
#require doesn't work in toplevel

### ISSUE THREAD

**Issue by yminsky** [janestreet/ppx_sexp_conv#30]
I can enable ppx_sexp_value if I #require "ppx_jane", but not if I #require "ppx_sexp_value".

```ocaml
utop # #require "ppx_jane";;
─( 19:25:31 )─< command 1 >─────────────────────────────────────────────────────────────{ counter: 0 }─
utop # open Base;;
─( 19:25:36 )─< command 2 >─────────────────────────────────────────────────────────────{ counter: 0 }─
utop # [%sexp_of: int];;
- : int -> Sexp.t = <fun>
```

```ocaml
utop # #require "ppx_sexp_value";;
─( 19:23:52 )─< command 1 >─────────────────────────────────────────────────────────────{ counter: 0 }─
utop # [%sexp_of: int];;
Line 1, characters 2-9:
Error: Uninterpreted extension 'sexp_of'.
```

Any idea what's going on?

**Comment by jeremiedimino** [comment]
This is expected. `ppx_sexp_conv` being a dependency is:
- a public fact of `ppx_jane`
- an implementation detail of `ppx_sexp_value`

We then carefully crafted the dependencies between these libraries to reflect this fact. So `ppx_sexp_value` not providing `[%sexp_of: ...]` is voluntary. You need to explicitly depend on `ppx_sexp_conv` to get access to it.

**Comment by yminsky** [comment]
Hmm.  I can't figure out how to enable [%sexp_of] or [@@deriving sexp] in utop except by #require "ppx_jane".  How should I do it?

**Comment by jeremiedimino** [comment]
It should work with: `#require "ppx_sexp_conv"`

**Comment by yminsky** [comment]
It doesn't seem to work for me:

```
utop # #require "base";;
─( 12:49:49 )─< command 1 >───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────{ counter: 0 }─
utop # #require "ppx_sexp_conv";;
No such package: ppx_deriving - required by `ppx_sexp_conv'
```

**Comment by jeremiedimino** [comment]
Ah, that's a deeper issue... For historical reasons, the toplevel relies on an older and more complicated way of composing and running ppx rewriters. Currently, you have to install the `ppx_deriving` package in order to use `[@@deriving]` plugins in the toplevel.

The reason it works with `ppx_jane` is a bit long to explain.

/cc @emillon, we should really look into changing this

**Comment by yminsky** [comment]
Is a reasonable short-term hack to make ppx_deriving a dependency of the various ppxs?  That way, when you install a PPX, you'll be able to actually use it in the toplevel without installing anything else.

**Comment by jeremiedimino** [comment]
Such changes never have a 0 cost as everybody cares about dependencies and we have been in this situations for years. So I'm more inclined to keep it as a know quirk until we switch to something more straightforward.

**Comment by yminsky** [comment]
Understood

**Comment by emillon** [comment]
> Ah, that's a deeper issue... For historical reasons, the toplevel relies on an older and more complicated way of composing and running ppx rewriters. Currently, you have to install the `ppx_deriving` package in order to use `[@@deriving]` plugins in the toplevel.
> 
> The reason it works with `ppx_jane` is a bit long to explain.
> 
> /cc @emillon, we should really look into changing this

Yeah, curious to hear why it works with `ppx_jane`. I suppose it's related to the fact that ppx_jane is installed as a precompiled driver usable with `#ppx` as described in the README.
